docs/cert-manager.md

```diff
@@ -1,76 +1,469 @@
+
+```shell
+console
+console.minio-operator.svc.
+console.minio-operator.svc.<cluster domain>
+```
+
+> [!IMPORTANT]
+> Replace `<cluster domain>` with the actual values for your MinIO tenant.
+> `cluster domain` is the internal root DNS domain assigned in your Kubernetes cluster. Typically this is `cluster.local`, check on your coredns
+> configuration for the correct value for your Kubernetes cluster. For example, using `kubectl get configmap coredns -n kube-system -oyaml | yq ".data"`.
+> The way the root DNS domain is managed can vary depending on the Kubernetes distribution (Openshift, Rancher, EKS, etc.)
+
+Create a `Certificate` for the domains mentioned above:
+
+```yaml
+# console-tls-certificate.yaml
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: console-certmanager-cert
+  namespace: minio-operator
+spec:
+  dnsNames:
+    - console
+    - console.minio-operator.svc
+    - console.minio-operator.svc.cluster.local
+  secretName: console-tls
+  issuerRef:
+    name: minio-operator-ca-issuer
+```
+
+```shell
+kubectl apply -f console-tls-certificate.yaml
+```
+
+This creates a secret called `console-tls` in the `minio-operator` namespace.
+
+> [!IMPORTANT]
+> The secret name is not optional. Make sure the secret name is `console-tls` by setting `spec.secretName: console-tls` as in the example above.
+
+### Install Operator with Auto TLS disabled for STS
+
+When installing the Operator deployment, make sure to set `OPERATOR_STS_AUTO_TLS_ENABLED: off` env variable in the `minio-operator` container. This prevents
+MinIO Operator from issuing the certificate for STS and instead wait for you to provide the TLS certificate issued by Cert Manager.
+
+Also make sure to set `MINIO_CONSOLE_TLS_ENABLE: off` env variable, this prevents MinIO Operator from issuing the certificate for Console
+and instead will expect you to provide the TLS certificate issued by Cert Manager.
+
+> [!WARNING]
+> Failing to provide the secret `sts-tls` containing the TLS certificate or providing an invalid key-pair in the secret will
+> prevent the STS service from starting.
+
+A way to make sure that the env variables are properly set is using kustomization to patch the `minio-operator` Deployment:
+
+```yaml
+# minio-operator/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- github.com/minio/operator/resources
+
+patches:
+- patch: |-
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: minio-operator
+      namespace: minio-operator
+    spec:
+      template:
+        spec:
+          containers:
+            - name: minio-operator
+              env:
+                - name: MINIO_CONSOLE_TLS_ENABLE
+                  value: "off"
+                - name: OPERATOR_STS_AUTO_TLS_ENABLED
+                  value: "off"
+                - name: OPERATOR_STS_ENABLED
+                  value: "on"
+```
+
+```shell
+kubectl apply -k minio-operator
```

Comment:
This did not work for me. This may be perhaps better:
```
kubectl patch deployment -n minio-operator minio-operator -p '{"spec":{"template":{"spec":{"containers":[{"name": "minio-operator", "env": [{"name": "OPERATOR_STS_AUTO_TLS_ENABLED", "value": "off"}, {"name": "MINIO_CONSOLE_TLS_ENABLE", "value": "off"}, {"name": "OPERATOR_STS_ENABLED", "value": "on"}]}]}}}}'
```

Reply:
This requires that you NOT have an operator deployment previously installed. If there is a previous deployment then the patch will go through, but it implies that the Operator deployment was created before, with the env variable OPERATOR_STS_AUTO_TLS_ENABLED default value as "ON", that is no good, because Operator is going to create CSR's and secretes before you are able to patch; we want to avoid that.